scripts/helper.sh

```diff
@@ -0,0 +1,213 @@
+    else
+        export "$var_name"="$var_value"
+        echo "Using value for $var_name: $var_value"
+    fi
+}
+
+# Check if a required parameter is set (either from env or command line)
+# Usage: check_param "PARAM_NAME" "$PARAM_VALUE"
+check_param() {
+    local param_name="$1"
+    local param_value="$2"
+    if [ -z "$param_value" ]; then
+        handle_error "$param_name is not set (check environment variables or command line parameters)"
+    else
+        export "$param_name"="$param_value"
+        echo "Using value for $param_name: $param_value"
+    fi
+}
+
+# Handle errors with a message and exit
+# Usage: handle_error "Error message"
+handle_error() {
+    local message="$1"
+    echo "Error: $message" >&2
+    exit 1
+}
+
+# Execute a transaction and handle errors
+# Usage: execute_transaction "description" "command"
+execute_transaction() {
+    local description="$1"
+    local command="$2"
+
+    echo "Executing: $description"
+    eval "$command"
+
+    if [ $? -eq 0 ]; then
+        echo "Successfully $description"
+    else
+        handle_error "Failed to $description"
+    fi
+}
+
+# Wait for Ethereum node to be ready
+# Usage: wait_for_ethereum
+wait_for_ethereum() {
+    echo "Waiting for Ethereum node to be ready..."
+    while ! curl -s -X POST -H "Content-Type: application/json" \
+                 --data '{"jsonrpc":"2.0","method":"net_version","params":[],"id":1}' \
+                 "$LOCAL_ETHEREUM_RPC_URL" > /dev/null 2>&1
+    do
+        echo "Waiting for $LOCAL_ETHEREUM_RPC_URL"
+        sleep 1
+    done
+    echo "Ethereum node is ready!"
+}
+
+# Impersonate an account (for local development)
+# Usage: impersonate_account "0x..."
+impersonate_account() {
+    local account="$1"
+    if [ "${DEPLOY_ENV:-}" = "TESTNET" ]; then
+        return 0
+    fi
+    
+    cast rpc anvil_impersonateAccount "$account" -r "$LOCAL_ETHEREUM_RPC_URL" > /dev/null 2>&1 || \
+        handle_error "Failed to impersonate account $account"
+    
+    cast rpc anvil_setBalance "$account" 0x10000000000000000000 -r "$LOCAL_ETHEREUM_RPC_URL" > /dev/null 2>&1 || \
+        handle_error "Failed to set balance for account $account"
+}
+
+# Stop impersonating an account (for local development)
+# Usage: stop_impersonating "0x..."
+stop_impersonating() {
+    local account="$1"
+    if [ "${DEPLOY_ENV:-}" = "TESTNET" ]; then
+        return 0
+    fi
+    
+    cast rpc anvil_stopImpersonatingAccount "$account" -r "$LOCAL_ETHEREUM_RPC_URL" > /dev/null 2>&1 || \
+        echo "Warning: Failed to stop impersonating account $account" >&2
+}
+
+# Set up environment based on DEPLOY_ENV
+# Usage: setup_environment
+setup_environment() { 
+    # Set up environment based on DEPLOY_ENV
+    if [ "$DEPLOY_ENV" = "LOCAL" ]; then
+        check_param "LOCAL_ETHEREUM_RPC_URL" "${LOCAL_ETHEREUM_RPC_URL:-http://localhost:8545}"
+    else
+        check_param "TESTNET_RPC_URL" "${TESTNET_RPC_URL:-}"
```

Comment:
why empty default here?

`${TESTNET_RPC_URL:-}` rather than `${TESTNET_RPC_URL}`

(You may have a good reason, it was just not obvious to me what the difference was, happy to hear the explanation)